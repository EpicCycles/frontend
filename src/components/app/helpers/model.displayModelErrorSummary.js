import {customerAddressFields} from "../model/helpers/fields";
import {displayModelErrorSummary} from "../model/helpers/model";

test('if no error or error_detail undefined returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(undefined);
});
test('if just error no error_detail error returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error: "Model level error"
    };
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe("Model level error");
});
test('if no error with error_detail error_details formatted returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error_detail: {
            "address1": ["A valid address is required.", "SHow me the address is required."],
            "postcode": ["This field may not be blank."]
        },
    };
    const expectedText = "Address: A valid address is required. SHow me the address is required.<br>Postcode: This field may not be blank.";
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(expectedText);
});
test('if error and error_detail formatted returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error: "Model level error",
        error_detail: {
            "address1": ["A valid address is required.", "SHow me the address is required."],
            "postcode": ["This field may not be blank."]
        },
    };
    const expectedText = "Model level error<br>Address: A valid address is required. SHow me the address is required.<br>Postcode: This field may not be blank.";
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(expectedText);
});