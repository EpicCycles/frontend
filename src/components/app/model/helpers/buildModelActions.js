/**
 * buildModelActions - builds actions for the current model (use to build additional actions)
 * @param requiredActions - array of objects {iconName, iconTitle, requiresIdOnly, actionFunction}
 * @param model - model being displayed oredited
 * @returns array of model actions specific to current model
 */
export const buildModelActions = (requiredActions, model) => {
  return requiredActions.map(action => {
    const finalAction = {
      iconName: action.iconName,
      iconTitle: action.iconTitle,
    };
    const isSavedModel = model && model.id;
    if (action.requiresIdOnly) {
      finalAction.iconAction = () => isSavedModel && action.actionFunction(model.id);
    } else {
      finalAction.iconAction = () => isSavedModel && action.actionFunction(model);
    }
    finalAction.iconDisabled = !isSavedModel;
    return finalAction;
  });
};
