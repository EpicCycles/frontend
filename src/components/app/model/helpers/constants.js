export const POSTCODE_RULES = [
  { countryCode: 'AT', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'AU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'BE', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'BG', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CA', regex: /^[A-Za-z][0-9][A-Za-z] [0-9][A-Za-z][0-9]/, display: 'ANA NAN' },
  { countryCode: 'CH', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CY', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CZ', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'DE', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'DK', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'EE', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'ES', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'FI', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'FR', regex: /^\d{5}$/, display: 'NNNNN' },
  {
    countryCode: 'GB',
    regex: /^[A-Z](([0-9][0-9]?)|([A-Z][0-9][0-9]?)|([A-Z]?[0-9][A-Z])) ?[0-9][ABD-HJLNP-UW-Z]{2}$/,
    display: 'CCNN NCC',
  },
  { countryCode: 'GR', regex: /^\d{3}\s\d{2}$/, display: 'NNNNN or NNN NN' },
  { countryCode: 'HU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'IT', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'KR', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'LT', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'LU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'MT', regex: /^\[A-Z]{3}\s\d{4}$/, display: 'AAA NNNN' },
  { countryCode: 'MY', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'NO', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'NZ', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'PH', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'PL', regex: /^\d{2}-\d{3}$/, display: 'NN-NNN' },
  { countryCode: 'PT', regex: /^\d{4}-\d{3}$/, display: 'NNNN-NNN' },
  { countryCode: 'RO', regex: /^\d{6}$/, display: 'NNNNNN' },
  { countryCode: 'SE', regex: /^\d{3,4}$/, display: 'NNNNN or NNNN' },
  { countryCode: 'SG', regex: /^\d{6}$/, display: 'NNNNNN' },
  { countryCode: 'SI', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'SK', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'US', regex: /^(\d{5}([-]\d{4})?)$/, display: 'NNNNN-NNNN or NNNNN' },
  { countryCode: 'ZA', regex: /^\d{4}$/, display: 'NNNN' },
];
