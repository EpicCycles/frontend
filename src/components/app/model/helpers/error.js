export const BRAND_NAME_MISSING = "A name is required for the brand";
export const FRAME_NAME_MISSING = "A name is required for the frame";
export const ATTRIBUTE_NAME_MISSING = "A name is required for the attribute";
export const BUNDLE_NAME_MISSING = "A name is required for the bundle";
export const MODEL_NAME_MISSING = "A name is required for the model";
export const PART_NAME_MISSING = "A name is required for the part";
export const PART_TYPE_NAME_MISSING = "A name is required for the part type";
export const SECTION_MISSING = "A section must be selected for the part type";
export const PART_TYPE_MISSING = "A part type must be selected";
export const BRAND_MISSING = "A brand must be selected";
export const PRODUCTS_MISSING = "At least 2 products must be selected";
export const SUPPLIER_MISSING = "A supplier must be selected";
export const PART_MISSING = "A part must be selected";
export const ADDRESS_MISSING = "At least one line of address must be entered";
export const COUNTRY_MISSING = "A country must be selected";
export const INVALID_POSTCODE = "A postcode is expected which for this country has the format ";
export const SELECT_ONE_MISSING = "Please select an option.";
export const VALUE_MISSING = "A value is required.";
export const INVALID_EMAIL = "Invalid email address.";
export const INVALID_URL = "Invalid URL";
