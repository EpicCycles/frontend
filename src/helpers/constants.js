export const NEW_ELEMENT_ID = 'new';

export const colourStyles = [
  { background: 'bg-col-01', colour: 'col-01', border: 'border-col-01' },
  {
    background: 'bg-col-02',
    colour: 'col-02',
    border: 'border-col-02',
    transition: 'transition-01-02',
  },
  {
    background: 'bg-col-03',
    colour: 'col-03',
    border: 'border-col-03',
    transition: 'transition-02-03',
  },
  {
    background: 'bg-col-04',
    colour: 'col-04',
    border: 'border-col-04',
    transition: 'transition-03-04',
  },
  {
    background: 'bg-col-05',
    colour: 'col-05',
    border: 'border-col-05',
    transition: 'transition-04-05',
  },
  {
    background: 'bg-col-06',
    colour: 'col-06',
    border: 'border-col-06',
    transition: 'transition-05-06',
  },
  {
    background: 'bg-col-07',
    colour: 'col-07',
    border: 'border-col-07',
    transition: 'transition-06-07',
  },
  {
    background: 'bg-col-08',
    colour: 'col-08',
    border: 'border-col-08',
    transition: 'transition-07-08',
  },
  {
    background: 'bg-col-09',
    colour: 'col-09',
    border: 'border-col-09',
    transition: 'transition-08-09',
  },
  {
    background: 'bg-col-10',
    colour: 'col-10',
    border: 'border-col-10',
    transition: 'transition-09-10',
  },
  {
    background: 'bg-col-11',
    colour: 'col-11',
    border: 'border-col-11',
    transition: 'transition-10-11',
  },
  {
    background: 'bg-col-12',
    colour: 'col-12',
    border: 'border-col-12',
    transition: 'transition-11-12',
  },
  {
    background: 'bg-col-13',
    colour: 'col-13',
    border: 'border-col-13',
    transition: 'transition-12-13',
  },
];

export const STORAGE_PARTS = 'epic_parts';
export const STORAGE_SUPPLIER_PRODUCTS = 'epic_supplierProducts';
export const STORAGE_SECTIONS = 'epic_sections';
export const STORAGE_BRANDS = 'epic_brands';
export const STORAGE_CHARGES = 'epic_charges';
export const STORAGE_SUPPLIERS = 'epic_suppliers';
export const COOKIE_USER = 'epic_user';
export const COOKIE_TOKEN = 'epic_user_token';
