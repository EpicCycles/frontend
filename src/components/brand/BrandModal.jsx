/* eslint-disable camelcase */
import React from 'react';

import * as PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react';
import { NEW_ELEMENT_ID } from '../../helpers/constants';
import { brandFields } from '../app/model/helpers/fields';
import EditModelPage from '../app/model/EditModelPage';
import { updateModel } from '../app/model/helpers/model';

class BrandModal extends React.Component {
  state = {};

  constructor(props) {
    super();
    this.state = this.deriveStateFromProps(props);
  }

  onAfterOpen = () => {
    this.setState(this.deriveStateFromProps(this.props));
  };

  deriveStateFromProps = props => {
    return {
      brand: props.brand || {},
      mode: props.brand && props.brand.id ? 'Edit' : 'New',
    };
  };

  handleBrandValueChange = (fieldName, input) => {
    const { componentKey, brand, suppliers } = this.props;

    const updatedBrand = updateModel(brand, brandFields, fieldName, input);
    if (fieldName.startsWith('supplier')) {
      updatedBrand.supplier_names = this.buildSupplierNameArray(input, suppliers);
    }

    if (componentKey === NEW_ELEMENT_ID) updatedBrand.dummyKey = NEW_ELEMENT_ID;

    this.setState({ brand: updatedBrand });
  };

  buildSupplierNameArray = (selectedSuppliers, suppliers) => {
    const supplier_names = [];
    suppliers.forEach(supplier => {
      if (selectedSuppliers.includes(supplier.id)) {
        supplier_names.push(supplier.supplier_name);
      }
    });
    return supplier_names;
  };

  onClickReset = () => {
    this.setState(this.deriveStateFromProps(this.props));
  };

  saveOrCreateBrand = () => {
    const { saveBrand, closeBrandModal } = this.props;
    const { brand } = this.state;
    saveBrand(brand);
    closeBrandModal();
  };

  deleteOrRemoveBrand = () => {
    const { deleteBrand, closeBrandModal } = this.props;
    const { componentKey, brand } = this.state;
    if (componentKey) {
      deleteBrand(brand);
    }
    closeBrandModal();
  };

  render() {
    const { brand, mode } = this.state;
    const { closeBrandModal, suppliers, brandModalOpen, deleteBrand } = this.props;
    // eslint-disable-next-line react/destructuring-assignment
    const persistedBrand = this.props.brand;
    return (
      <div>
        <div style={{ width: '100%', textAlign: 'right' }}>
          <Icon name="remove" circular link onClick={closeBrandModal} />
        </div>
        <div style={{ width: '100%', textAlign: 'left' }}>
          <h2>{mode} Brand</h2>
          <EditModelPage
            model={brand}
            persistedModel={persistedBrand}
            modelFields={brandFields}
            onChange={this.handleBrandValueChange}
            suppliers={suppliers}
          />
        </div>
        <div style={{ width: '100%', textAlign: 'right' }}>
          {brand.changed && (
            <Icon
              id="reset-brand"
              name="undo"
              onClick={this.onClickReset}
              title="Reset Brand details"
            />
          )}
          {brand.changed && !brand.error && (
            <Icon
              id="accept-brand"
              name="check"
              onClick={this.saveOrCreateBrand}
              title="Confirm Brand Change"
            />
          )}
          {deleteBrand && (brand.id || brand.changed) && (
            <Icon
              id="delete-brand"
              name="trash"
              onClick={this.deleteOrRemoveBrand}
              title="Delete Brand"
            />
          )}
        </div>
      </div>
    );
  }
}

BrandModal.propTypes = {
  brandModalOpen: PropTypes.bool.isRequired,
  brand: PropTypes.object.isRequired,
  componentKey: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  suppliers: PropTypes.array.isRequired,
  saveBrand: PropTypes.func.isRequired,
  deleteBrand: PropTypes.func.isRequired,
  closeBrandModal: PropTypes.func.isRequired,
};

export default BrandModal;
