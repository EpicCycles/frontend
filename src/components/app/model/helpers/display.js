import {
    BIKE,
    BRAND,
    CHECKBOX,
    CURRENCY,
    CUSTOMER,
    DATE_TIME,
    NUMBER,
    PART_TYPE,
    RADIO,
    SELECT_ONE,
    SUPPLIER,
    USER
} from "./fields";
import {getCustomerName} from "../../../customer/helpers/customer";
import {getBrandName} from "../../../brand/helpers/brand";
import {getNameForValue} from "./model";
import {getSupplierName} from "../../../supplier/helpers/supplier";
import {getPartTypeName} from "../../../partType/helpers/partType";
import {getBikeName} from "../../../bike/helpers/bike";
import {getUserName} from "../../../user/helpers/user";

export const fixedHeaderClassname = (lockColumn) => {
    if (lockColumn) return "grid-header--fixed-left";
    return "";
};
export const fixedDetailsClassname = (lockColumn) => {
    if (lockColumn) return "grid-item--fixed-left";
    return "";
};
export const gridHeaderClass = (baseClassName = "", fieldIndex, firstColumnLocked) => {
    const shouldLock = firstColumnLocked && (fieldIndex === 0);
    return `${baseClassName} grid-item--header ${fixedHeaderClassname(shouldLock)}`;
};
export const gridItemClass = (baseClassName = "", fieldIndex, firstColumnLocked) => {
    const shouldLock = firstColumnLocked && (fieldIndex === 0);
    return `${baseClassName} grid-item ${fixedDetailsClassname(shouldLock)}`;
};
export const formattedDate = date => {
    if (date) return date.toLocaleDateString('en-GB');
    return "";
};
export const formattedDateTime = date => {
    if (date) return date.toLocaleString('en-GB');
    return "";
};
export const fieldAlignment = field => {
    switch (field.type) {
        case NUMBER:
        case CURRENCY:
            return 'align_right';
        case CHECKBOX:
        case RADIO:
            return 'align_center';
        default:
            return '';
    }
};
export const buildViewString = (model, field, sections, brands, suppliers, customers, bikes, frames, users) => {
    let viewData;
    const fieldValue = model ? model[field.fieldName] : undefined;
    switch (field.type) {
        case CUSTOMER:
            viewData = fieldValue ? getCustomerName(fieldValue, customers) : '';
            break;
        case CURRENCY:
            viewData = fieldValue ? Number(fieldValue).toLocaleString('en-GB', {
                style: 'currency',
                currency: 'GBP'
            }) : "";
            break;
        case CHECKBOX:
            viewData = fieldValue ? "Y" : "N";
            break;
        case DATE_TIME:
            viewData = fieldValue ? formattedDateTime(new Date(fieldValue)) : "";
            break;
        case PART_TYPE:
            viewData = fieldValue ? getPartTypeName(fieldValue, sections) : '';
            break;
        case BRAND:
            viewData = fieldValue ? getBrandName(fieldValue, brands) : '';
            break;
        case SUPPLIER:
            viewData = fieldValue ? getSupplierName(fieldValue, suppliers) : '';
            break;
        case BIKE:
            viewData = fieldValue ? getBikeName(fieldValue, bikes, frames, brands) : '';
            break;
        case SELECT_ONE:
            viewData = fieldValue ? getNameForValue(fieldValue, field.selectList) : '';
            break;
        case USER:
            viewData = fieldValue ? getUserName(fieldValue, users) : '';
            break;
        default:
            viewData = fieldValue ? fieldValue : '';
    }
    return viewData;
};
