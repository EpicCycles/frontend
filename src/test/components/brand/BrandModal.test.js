import React from 'react';
import toJson from 'enzyme-to-json';
import BrandModal from '../../../components/brand/BrandModal';
import { BRAND_NAME_MISSING } from '../../../components/app/model/helpers/error';
import { NEW_ELEMENT_ID } from '../../../helpers/constants';

const suppliers = [
  { id: 1, supplier_name: 'supplier 1' },
  { id: 2, supplier_name: 'supplier 2' },
  { id: 3, supplier_name: 'supplier 3' },
  { id: 4, supplier_name: 'supplier 4' },
];

describe('BrandModal', () => {
  it('BrandModal displays correctly for a new brand', () => {
    const component = shallow(
      <BrandModal
        brand={{}}
        componentKey={NEW_ELEMENT_ID}
        brandModalOpen
        deleteBrand={jest.fn()}
        saveBrand={jest.fn()}
        closeBrandModal={jest.fn()}
        suppliers={suppliers}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it('BrandModal displays correctly for a brand', () => {
    const brand = {
      brand_name: 'e brand 8',
      link: 'https://bianchi.co.uk',
      id: 8,
      supplier: [23, 1, 2, 45, 16],
      supplier_names: ['name 23', 'name 1', 'name 2', 'name 45', 'name 16'],
    };
    const componentKey = brand.id;
    const component = shallow(
      <BrandModal
        brandModalOpen
        brand={brand}
        componentKey={componentKey}
        suppliers={suppliers}
        deleteBrand={jest.fn()}
        saveBrand={jest.fn()}
        closeBrandModal={jest.fn()}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it('BrandModal value changes processing 1', () => {
    const brand = {
      brand_name: 'e brand 8',
      link: 'https://bianchi.co.uk',
      id: 8,
      supplier: [23, 1, 2, 45, 16],
      supplier_names: ['name 23', 'name 1', 'name 2', 'name 45', 'name 16'],
    };
    const componentKey = brand.id;
    const component = shallow(
      <BrandModal
        brandModalOpen
        brand={brand}
        componentKey={componentKey}
        suppliers={suppliers}
        deleteBrand={jest.fn()}
        saveBrand={jest.fn()}
        closeBrandModal={jest.fn()}
      />,
    );

    // brand name change to no value
    component.instance().handleBrandValueChange('brand_name', '');
    const expectedData = Object.assign({}, brand, {
      brand_name: null,
      changed: true,
      error_detail: { brand_name: BRAND_NAME_MISSING },
    });
    expect(component.state('brand')).toEqual(expectedData);
  });
  it('BrandModal value changes processing 2', () => {
    const brand = {
      brand_name: 'e brand 8',
      link: 'https://bianchi.co.uk',
      id: 8,
      supplier: [23, 1, 2, 45, 16],
      supplier_names: ['name 23', 'name 1', 'name 2', 'name 45', 'name 16'],
    };
    const componentKey = brand.id;
    const component = shallow(
      <BrandModal
        brandModalOpen
        brand={brand}
        componentKey={componentKey}
        suppliers={suppliers}
        deleteBrand={jest.fn()}
        saveBrand={jest.fn()}
        closeBrandModal={jest.fn()}
      />,
    );

    // brand name change to value
    component.instance().handleBrandValueChange('brand_name', 'new brand name');
    const expectedData = Object.assign({}, brand, {
      brand_name: 'new brand name',
      changed: true,
      error_detail: {},
    });
    expect(component.state('brand')).toEqual(expectedData);
  });
  it('BrandModal value changes processing 3', () => {
    const brand = {
      brand_name: 'e brand 8',
      link: 'https://bianchi.co.uk',
      id: 8,
      supplier: [23, 1, 2, 45, 16],
      supplier_names: ['name 23', 'name 1', 'name 2', 'name 45', 'name 16'],
    };
    const componentKey = brand.id;
    const component = shallow(
      <BrandModal
        brandModalOpen
        brand={brand}
        componentKey={componentKey}
        suppliers={suppliers}
        deleteBrand={jest.fn()}
        saveBrand={jest.fn()}
        closeBrandModal={jest.fn()}
      />,
    );
    // link change to no value
    component.instance().handleBrandValueChange('link', '');
    const expectedData = Object.assign({}, brand, { link: null, changed: true, error_detail: {} });
    expect(component.state('brand')).toEqual(expectedData);
  });
  it('BrandModal value changes processing 4', () => {
    const brand = {
      brand_name: 'e brand 8',
      link: 'https://bianchi.co.uk',
      id: 8,
      supplier: [23, 1, 2, 45, 16],
      supplier_names: ['name 23', 'name 1', 'name 2', 'name 45', 'name 16'],
    };
    const componentKey = brand.id;
    const component = shallow(
      <BrandModal
        brandModalOpen
        brand={brand}
        componentKey={componentKey}
        suppliers={suppliers}
        deleteBrand={jest.fn()}
        saveBrand={jest.fn()}
        closeBrandModal={jest.fn()}
      />,
    );

    // link change to new value
    component.instance().handleBrandValueChange('link', 'myLink.co.uk');
    const expectedData = Object.assign({}, brand, {
      link: 'myLink.co.uk',
      changed: true,
      error_detail: {},
    });
    expect(component.state('brand')).toEqual(expectedData);
  });
  it('BrandModal value changes processing 5', () => {
    const brand = {
      brand_name: 'e brand 8',
      link: 'https://bianchi.co.uk',
      id: 8,
      supplier: [23, 1, 2, 45, 16],
      supplier_names: ['name 23', 'name 1', 'name 2', 'name 45', 'name 16'],
    };
    const componentKey = brand.id;
    const component = shallow(
      <BrandModal
        brandModalOpen
        brand={brand}
        componentKey={componentKey}
        suppliers={suppliers}
        deleteBrand={jest.fn()}
        saveBrand={jest.fn()}
        closeBrandModal={jest.fn()}
      />,
    );

    // supplier list changes
    component.instance().handleBrandValueChange('supplier', [suppliers[1].id]);
    const expectedData = Object.assign({}, brand, {
      supplier: [suppliers[1].id],
      supplier_names: [suppliers[1].supplier_name],
      changed: true,
      error_detail: {},
    });
    expect(component.state('brand')).toEqual(expectedData);
  });
});
