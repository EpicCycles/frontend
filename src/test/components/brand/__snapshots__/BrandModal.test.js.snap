// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandModal BrandModal displays correctly for a brand 1`] = `
<div>
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  >
    <Icon
      as="i"
      circular={true}
      link={true}
      name="remove"
      onClick={[MockFunction]}
    />
  </div>
  <div
    style={
      Object {
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <h2>
      Edit
       Brand
    </h2>
    <EditModelPage
      model={
        Object {
          "brand_name": "e brand 8",
          "id": 8,
          "link": "https://bianchi.co.uk",
          "supplier": Array [
            23,
            1,
            2,
            45,
            16,
          ],
          "supplier_names": Array [
            "name 23",
            "name 1",
            "name 2",
            "name 45",
            "name 16",
          ],
        }
      }
      modelFields={
        Array [
          Object {
            "error": "A name is required for the brand",
            "fieldName": "brand_name",
            "header": "Brand Name",
            "length": 50,
            "required": true,
            "size": 20,
            "synonyms": Array [],
          },
          Object {
            "fieldName": "bike_brand",
            "header": "Bike Brand",
            "type": "checkbox",
          },
          Object {
            "fieldName": "link",
            "header": "URL",
            "length": 100,
            "size": 40,
            "type": "text",
            "validator": [Function],
          },
          Object {
            "fieldName": "supplier",
            "header": "Supplier",
            "synonyms": Array [
              "supplier",
            ],
            "type": "supplier",
          },
        ]
      }
      onChange={[Function]}
      persistedModel={
        Object {
          "brand_name": "e brand 8",
          "id": 8,
          "link": "https://bianchi.co.uk",
          "supplier": Array [
            23,
            1,
            2,
            45,
            16,
          ],
          "supplier_names": Array [
            "name 23",
            "name 1",
            "name 2",
            "name 45",
            "name 16",
          ],
        }
      }
      suppliers={
        Array [
          Object {
            "id": 1,
            "supplier_name": "supplier 1",
          },
          Object {
            "id": 2,
            "supplier_name": "supplier 2",
          },
          Object {
            "id": 3,
            "supplier_name": "supplier 3",
          },
          Object {
            "id": 4,
            "supplier_name": "supplier 4",
          },
        ]
      }
    />
  </div>
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  >
    <Icon
      as="i"
      id="delete-brand"
      name="trash"
      onClick={[Function]}
      title="Delete Brand"
    />
  </div>
</div>
`;

exports[`BrandModal BrandModal displays correctly for a new brand 1`] = `
<div>
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  >
    <Icon
      as="i"
      circular={true}
      link={true}
      name="remove"
      onClick={[MockFunction]}
    />
  </div>
  <div
    style={
      Object {
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <h2>
      New
       Brand
    </h2>
    <EditModelPage
      model={Object {}}
      modelFields={
        Array [
          Object {
            "error": "A name is required for the brand",
            "fieldName": "brand_name",
            "header": "Brand Name",
            "length": 50,
            "required": true,
            "size": 20,
            "synonyms": Array [],
          },
          Object {
            "fieldName": "bike_brand",
            "header": "Bike Brand",
            "type": "checkbox",
          },
          Object {
            "fieldName": "link",
            "header": "URL",
            "length": 100,
            "size": 40,
            "type": "text",
            "validator": [Function],
          },
          Object {
            "fieldName": "supplier",
            "header": "Supplier",
            "synonyms": Array [
              "supplier",
            ],
            "type": "supplier",
          },
        ]
      }
      onChange={[Function]}
      persistedModel={Object {}}
      suppliers={
        Array [
          Object {
            "id": 1,
            "supplier_name": "supplier 1",
          },
          Object {
            "id": 2,
            "supplier_name": "supplier 2",
          },
          Object {
            "id": 3,
            "supplier_name": "supplier 3",
          },
          Object {
            "id": 4,
            "supplier_name": "supplier 4",
          },
        ]
      }
    />
  </div>
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  />
</div>
`;
